tlatools/org.lamport.tlatools/src/util/Resolver.java

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2025, Oracle and/or its affiliates.
+
+package util;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Optional;
+
+/**
+ * Resolves abstract filenames (e.g. <code>"Integers.tla"</code>) to concrete locations
+ * (e.g. <code>"jar:file:/path/to/tla2tools.jar!/tla2sany/StandardModules/Integers.tla"</code>).
+ *
+ * <p>This interface is simple enough to be composable: multiple resolvers can be chained together
+ * using {@link SequentialResolver}.
+ */
+interface Resolver {
+
+    /**
+     * Locate the given file.
```

Comment:
Add a note emphasizing that this function expects `ModuleName.tla`, not `ModuleName`.